api/src/services/tus/server.ts

```diff
@@ -88,8 +99,29 @@ export async function createTusServer(context: Context): Promise<[Server, () =>
 					tus_id: null,
 					tus_data: null,
 				});
+
+				fileData = {
+					...file,
+					...metadata,
+					tus_id: null,
+					tus_data: null,
+				};
 			}
 
+			emitter.emitAction(
+				'files.upload',
+				{
+					payload: fileData,
```

Comment:
With the way we are building the fileData this will bypass any `hooks` that may be mutating the data before updating via `updateOne`